### PR TITLE
Ruletest Engine update: Manual check, undetermined, and not applicable updates

### PR DESCRIPTION
**Pull request by jugonzal07**
Made updates to existing engine to account for results other than 'pass' and 'fail'.

### CODE REVIEW

**Comment by jugonzal07** [comment]
Looks like I may have mixed in the master Lighting test JSON in the process as well-- apologies for that. It shouldn't hurt anything and is up-to-date with the latest lighting master spreadsheet.

File: rct229/ruletest_engine/ruletest_engine.py [hunk 1]
```diff
@@ -82,23 +83,23 @@ def evaluate_outcome_enumeration_str(outcome_enumeration_str):
...
-        test_result = False
+        test_result = "fail"
     elif outcome_enumeration_str == "UNDETERMINED":
-        test_result = False
-    elif outcome_enumeration_str == "MISSING_CONTEXT":
-        test_result = False
-    elif outcome_enumeration_str == "NA":
-        test_result = False
+        test_result = "undetermined"
+    elif outcome_enumeration_str == "REQUIRES_MANUAL_CHECK":
```
Comment (weilixu) [inline rct229/ruletest_engine/ruletest_engine.py:99]: The Manual Check flag is a sub category of UNDETERMINED. So the outcome enumeration should only be PASSED, FAILED, UNDETERMINED and NOT_APPLICABLE

**Review by weilixu — APPROVED** [review]
Approved with a minor comment

**Review by yunjoonjung-PNNL — APPROVED** [review]
It looks great to me. Thank you Juan.

File: rct229/ruletest_engine/ruletest_engine.py (hunk 1 above)
Comment (jugonzal07) [reply rct229/ruletest_engine/ruletest_engine.py]: Thanks! This has been addressed in the latest push for this branch.

**Comment by jugonzal07** [comment]
All fixes addressed. Should be ready to merge.

**Pull request merged**